### PR TITLE
Drastic price change

### PR DESCRIPTION
**Pull request by dikel**

### CODE REVIEW

File: drastic-price-change-template/README.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# Drastic Price Change Anomaly Detection Bot
+
+## Description
+
+This bot detects if the price of an asset changes drastically or there is a large discrepancy between an on-chain and an off-chain oracle 
+
+## Supported Chains
+
+- Ethereum (For now)
```
Comment (christian-forta) [inline drastic-price-change-template/README.md]: fine for now, but we should expand it to all chains

File: drastic-price-change-template/README.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  - Type is always set to "suspicious"
+
+- PRICE-FLUCTUATIONS
+  - Fired when the price of an asset changes drastically
+  - Severity is always set to "low"
+  - Type is always set to "suspicious"
+
+## Test Data
+
+The bot behaviour can be verified with the provided unit tests
```
Comment (christian-forta) [inline drastic-price-change-template/README.md:33]: i think inverse finance would be a good one to validate this on

Comment (christian-forta) [reply drastic-price-change-template/README.md:33]: as well as LUNA token (chainlink circuit broke their oracle as soon as the price dropped below 0.1 whereas on centralized exchanges it was well below that)

File: drastic-price-change-template/bot-config.json.example [hunk 3]
```diff
@@ -0,0 +1,8 @@
+{
+  "priceDiscrepancyThreshold": 10,
+  "asset": {
+    "contract": "0x1f9840a85d5af5bf1d1762f925bdaddc4201f984",
```
Comment (christian-forta) [inline drastic-price-change-template/bot-config.json.example:4]: what is this?

File: drastic-price-change-template/bot-config.json.example [hunk 4]
```diff
@@ -0,0 +1,8 @@
+{
+  "priceDiscrepancyThreshold": 10,
```
Comment (christian-forta) [inline drastic-price-change-template/bot-config.json.example:2]: Not sure what the 10 represents? Is this for on-chain/off-chain discrepancies?

File: drastic-price-change-template/src/agent.js [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+    // Check the price once every INTERVAL
+    if (timestamp < lastTimestamp + INTERVAL) return findings;
+
+    // Get the price from chainlink and coingecko
+    const [chainlinkPrice, coingeckoPrice] = await Promise.all([
+      getChainlinkPriceFn(chainlinkContract, assetDecimals),
+      getCoingeckoPriceFn(asset.coingeckoId),
+    ]);
+
+    if (calculatePercentage(chainlinkPrice, coingeckoPrice) >= priceDiscrepancyThreshold) {
```
Comment (christian-forta) [inline drastic-price-change-template/src/agent.js:124]: how would we handle simple lags in price information during volatile times?

**Review by christian-forta — CHANGES_REQUESTED** [review]

File: drastic-price-change-template/bot-config.json.example (hunk 4 above)
Comment (dikel) [reply drastic-price-change-template/bot-config.json.example:2]: Yes, it is

File: drastic-price-change-template/src/agent.js (hunk 5 above)
Comment (dikel) [reply drastic-price-change-template/src/agent.js:124]: The chainlink price feeds can be updated when enough time has passed or if a "Deviation threshold" is met. For most of the assets I checked it is <2% so the chainlink price should be update on time

File: drastic-price-change-template/bot-config.json.example (hunk 3 above)
Comment (dikel) [reply drastic-price-change-template/bot-config.json.example:4]: This is the token contract. We use it to get all transfers from/to it. In this case this is the UNI token

File: drastic-price-change-template/README.md (hunk 2 above)
Comment (dikel) [reply drastic-price-change-template/README.md:33]: Chainlink no longer provide price feed for LUNA, I used the wayback machine to get the old contract, but it is LUNA/ETH and we would have to modify the bot a lot to work with ETH oracle so I don't think it is worth testing

File: drastic-price-change-template/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,9 @@
+FROM node:12-alpine
```
Comment (notyouraveragedev) [inline drastic-price-change-template/Dockerfile]: Node version

File: drastic-price-change-template/src/agent.js [hunk 7]
```diff
@@ -0,0 +1,202 @@
...
+    await Promise.all(uniqueAddresses.map(async (address) => {
+      if (!tokenHolders[address]) {
+        const balance = await contract.balanceOf(
+          address,
+          { blockTag: txEvent.blockNumber - 1 },
+        );
+        const balanceNormalized = Number(ethers.utils.formatUnits(balance, assetDecimals));
+        tokenHolders[address] = balanceNormalized;
+      }
+    }));
```
Comment (notyouraveragedev) [inline drastic-price-change-template/src/agent.js:87]: Wouldn't it be better to use multicall?

**Review by notyouraveragedev — CHANGES_REQUESTED** [review]

**Review by stoqnkpL — APPROVED** [review]
LGTM

**Pull request merged**